File: C/C-Sheet #2-Loop/H.c
#include <stdio.h>

int main() {
  int num,res,flag=0;
  scanf("%d",&num);
  res = num/2 ;
  
  for (int i = 2 ; i <= res; i++){
      if(num % i == 0){
        printf("NO\n");
        flag=1;
        break;
      }

    }
    if(flag == 0){
          printf("YES\n");
    }

  return 0;
}

File: Problem Solving/CodeForces/ICPC Assiut University Training Sheet/Newcomers/Java/C-Sheet #2-Loop/H.java
import java.util.Scanner;

public class H {
  public static void main(String[] args){
  Scanner input = new Scanner(System.in);
  int num,res,flag=0;
  num = input.nextInt();
  res = num/2 ;
  
  for (int i = 2 ; i <= res; i++){
      if(num % i == 0){
        System.out.println("NO");
        flag=1;
        break;
      }

    }
    if(flag == 0){
      System.out.println("YES");
    }
    input.close();
  }
}